### PR TITLE
[6.x] Remove calls to TestCase::at

### PR DESCRIPTION
**Pull request by soilSpoon**
`TestCase::at` is deprecated as of PHPUnit 9
So I refactor the test.

```
❯ ./vendor/bin/phpunit --stop-on-warning
PHPUnit 9.4.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.13
Configuration: /Users/dh/Projects/framework/phpunit.xml.dist

.............................................................   61 / 5729 (  1%)
.............................................................  122 / 5729 (  2%)
.............................................................  183 / 5729 (  3%)
.............................................................  244 / 5729 (  4%)
.............................................................  305 / 5729 (  5%)
.............................................................  366 / 5729 (  6%)
.............................................................  427 / 5729 (  7%)
.........SSSSSSSSSSSS........................................  488 / 5729 (  8%)
...................................SSSSSS....................  549 / 5729 (  9%)
...................................S.S.......................  610 / 5729 ( 10%)
.............................................................  671 / 5729 ( 11%)
.............................................................  732 / 5729 ( 12%)
....................................................W

Time: 00:00.662, Memory: 140.50 MB

There was 1 warning:

1) Illuminate\Tests\Database\DatabaseConnectionTest::testBeginTransactionMethodRetriesOnFailure
The at() matcher has been deprecated. It will be removed in PHPUnit 10. Please refactor your test to not rely on the order in which methods are invoked.

```

## Reference
sebastianbergmann/phpunit#4297

### CODE REVIEW

**Comment by driesvints** [comment]
There's still one in the `ValidationValidatorTest` it seems. Can you patch that one as well? https://github.com/laravel/framework/pull/35474/checks?check_run_id=1493278872#step:8:96

**Comment by soilSpoon** [comment]
@driesvints Oh my mistake, thank you for finding me

File: tests/Validation/ValidationValidatorTest.php [hunk 1]
```diff
@@ -1787,15 +1787,13 @@ public function testValidateMax()
...
+        $file->method('getSize')->willReturn(4096);
+
+        $v = new Validator($trans, ['photo' => $file], ['photo' => 'Max:4']);
         $this->assertTrue($v->passes());
 
-        $file = $this->getMockBuilder(UploadedFile::class)->setMethods(['isValid', 'getSize'])->setConstructorArgs([__FILE__, basename(__FILE__)])->getMock();
-        $file->expects($this->at(0))->method('isValid')->willReturn(true);
-        $file->expects($this->at(1))->method('getSize')->willReturn(4072);
-        $v = new Validator($trans, ['photo' => $file], ['photo' => 'Max:2']);
+        $v = new Validator($trans, ['photo' => $file], ['photo' => 'Max:3']);
```
Comment (driesvints) [inline tests/Validation/ValidationValidatorTest.php:1796]: Can you please revert the changes to the behavior of the tests here? Just focus on the `at` method.

Comment (soilSpoon) [reply tests/Validation/ValidationValidatorTest.php:1796]: Sorry. It seemed like it could be tested with one file, so I did this. I'll put it back

File: tests/Validation/ValidationValidatorTest.php [hunk 2]
```diff
@@ -1787,15 +1787,15 @@ public function testValidateMax()
...
+        $v = new Validator($trans, ['photo' => $file], ['photo' => 'Max:4']);
         $this->assertTrue($v->passes());
 
         $file = $this->getMockBuilder(UploadedFile::class)->setMethods(['isValid', 'getSize'])->setConstructorArgs([__FILE__, basename(__FILE__)])->getMock();
-        $file->expects($this->at(0))->method('isValid')->willReturn(true);
-        $file->expects($this->at(1))->method('getSize')->willReturn(4072);
-        $v = new Validator($trans, ['photo' => $file], ['photo' => 'Max:2']);
+        $file->method('isValid')->willReturn(true);
+        $file->method('getSize')->willReturn(4096);
+        $v = new Validator($trans, ['photo' => $file], ['photo' => 'Max:3']);
```
Comment (driesvints) [inline tests/Validation/ValidationValidatorTest.php:1798]: The original values are still not reverted. Please revert those as well.

**Review by driesvints — APPROVED** [review]

**Comment by driesvints** [comment]
@soilSpoon thanks!

**Review by GrahamCampbell — APPROVED** [review]
